src/main/java/com/metamx/metrics/JvmMonitor.java

```diff
@@ -30,8 +34,27 @@
 
 public class JvmMonitor extends AbstractMonitor
 {
-  private final KeyedDiff gcDiff = new KeyedDiff();
-
+  private final Sigar sigar = SigarUtil.getSigar();
+  /**
+   * The following code is partially based on
+   * https://github.com/aragozin/jvm-tools/blob/e0e37692648951440aa1a4ea5046261cb360df70/
+   * sjk-core/src/main/java/org/gridkit/jvmtool/PerfCounterGcCpuUsageMonitor.java
+   *
+   * Names of counters could also be seen at
+   * http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/be698ac28848/
+   * src/share/classes/sun/tools/jstat/resources/jstat_options
+   */
+  private final JStatData jStatData = JStatData.connect(sigar.getPid());
```

Comment:
Yes, server-metrics itself not a hotspot, but JStatData looks for me as a black box and I don't know how `connect` method affects entire JVM process. Maybe it adds some overhead on methods call (as for example profiles) or GC, that's why I'm asking how it works to make sure that we don't affect main JVM process significantly.